cpp-terminal/terminal.h

```diff
@@ -99,6 +97,16 @@ inline std::string cursor_on()
     return "\x1b[?25h";
 }
 
+inline char ctrl_key(unsigned char k)
+{
+    return k & 0x1f;
+}
+
+inline char alt_key(unsigned char k)
+{
+    return k + 0x80;
+}
+
```

Comment:
You can perhaps try `constexpr`, that might allow you to use it in the switch. I assume the constant must be known at compile time, this might allow it to do it.